app/jobs/bulk_api_job.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+class BulkApiJob < ApplicationJob
+  class Error < StandardError
+    class JobAlreadyRunning < Error
+      def message
+        "User is already running a bulk API job. Users can only run one bulk job at a time."
+      end
+    end
+
+    class MissingUser < Error
+      def message
+        "First argument in perform method must be the User performing the job."
+      end
+    end
+  end
+
+  queue_as :bulk_api_job
+
+  rescue_from(GitHub::Forbidden, Error::JobAlreadyRunning) do
+    user = arguments.first
```

Comment:
this restricts the first argument to be a `User`, maybe we could consider using keyword arguments, so we are not dependent on `.last` and `.first` methods?